smartlearner/utils.py

```diff
@@ -17,14 +18,41 @@ def sharedX(value, name=None, borrow=True, keep_on_cpu=False):
                          borrow=borrow)
 
 
+class NumpyEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, np.ndarray):
+            return {"__ndarray__": obj.tolist()}
+
+        return json.JSONEncoder(self, obj)
+
+
+def json_numpy_obj_hook(dct):
+    if isinstance(dct, dict) and '__ndarray__' in dct:
+        return np.array(dct['__ndarray__'])
+
+    return dct
+
+
 def save_dict_to_json_file(path, dictionary):
```

Comment:
Yes, they are used. For instance: https://github.com/SMART-Lab/smartlearner/pull/62/files#diff-8ebe7f074fcf17a46f8e635294cf4782R31 and https://github.com/SMART-Lab/smartlearner/pull/62/files#diff-8ebe7f074fcf17a46f8e635294cf4782R34